### PR TITLE
Various constructor and initialization improvements.

### PR DESCRIPTION
**Pull request by Rakete1111**
I removed one `std::cout` call, I hope that's ok, because now the initialization happens in the constructors, so a log call is impossible AFAIK.

### CODE REVIEW

**Comment by saurabhshri** [comment]
The PR looks perfect :) There's only a little change remaining.

Looks like the change in localTime population isn't working as expected in macOS at least. 

```
logPath             : tempFiles/03-12-2017-16-40-26.log
phonemeLogPath      : tempFiles/phoneme-03-12-2017-16-40-26.log
```

screen shot 2017-12-03 at 4 44 46 pm

This portion was just fixed yesterday because the results varied across Windows & Unix (macOS). We decided not to mix C++ strings and C strings. Since strftime works with NUL terminated strings, it made sense to allow it to use that.

File: src/lib_ccaligner/params.h [hunk 1]
```diff
@@ -9,11 +9,13 @@
 
 #include "commons.h"
 
-std::string currentTime();
+#include <cstring>
```
Comment (saurabhshri) [inline src/lib_ccaligner/params.h]: Is it really needed? ;)

Comment (Rakete1111) [reply src/lib_ccaligner/params.h]: @saurabhshri No ;) It should have been `ctime`...

**Comment by Rakete1111** [comment]
@saurabhshri Now I understand why you used C-style strings! `std::string` actually works perfectly, the only problem is that before, we didn't remove the trailing null characters from the string that appeared when resizing. Those <?> are null characters! So, we have to remove them and it should work. I don't have a mac though, so please test :)

**Comment by saurabhshri** [comment]
@Rakete1111 Thank you for such an amazing work! :)

**Pull request merged**